internal/activity.go

```diff
@@ -375,3 +382,18 @@ func newActivityContext(
 
 	return ctx, nil
 }
+
+func calculateActivityDeadline(scheduled, started time.Time, scheduleToCloseTimeout, startToCloseTimeout time.Duration) time.Time {
+	startToCloseDeadline := started.Add(startToCloseTimeout)
+	if scheduleToCloseTimeout > 0 {
+		scheduleToCloseDeadline := scheduled.Add(scheduleToCloseTimeout)
+		// Minimum of the two deadlines.
+		if scheduleToCloseDeadline.Before(startToCloseDeadline) {
+			return scheduleToCloseDeadline
+		} else {
+			return startToCloseDeadline
+		}
+	} else {
+		return startToCloseDeadline
+	}
```

Comment:
```suggestion
		if scheduleToCloseDeadline.Before(startToCloseDeadline) {
			return scheduleToCloseDeadline
		}
	}
	return startToCloseDeadline
```

Just reads clearer to me personally